common/lib/xmodule/xmodule/modulestore/mixed.py

```diff
@@ -172,8 +172,12 @@ def _get_modulestore_for_courseid(self, course_id=None):
             if mapping is not None:
                 return mapping
             else:
+                if isinstance(course_id, LibraryLocator):
```

Comment:
Yes, agreed about the need for a new base key type - that [was always my plan](https://openedx.atlassian.net/wiki/display/SOL/Content+Libraries#ContentLibraries-NewLocators).

I like `_get_modulestore_for_courselike(locator)` for now - I can make that change.